xclim/indices/_agro.py

```diff
@@ -672,6 +673,159 @@ def water_budget(
+
+    pr: xr.DataArray
+        Precipitation data.
+    s_thresh_wet: str
+        Accumulated precipitation threshold associated with `s_window_wet`.
+    s_window_wet: int
+        Number of days when accumulated precipitation is above `s_thresh_wet`. Defines the first condition to start the rain season
+    s_window_not_dry: int
+        Number of days, after `s_window_wet` days, during which no dry period must be found as a second and last condition to start the rain season.
+        A dry sequence is defined with `s_thresh_dry`, `s_window_dry` and `s_method_dry`.
+    s_thresh_dry: str
+        Threshold defining the sequence of dry days related to `s_window_dry`.
+    s_window_dry: int
+        Number of days used to define a dry sequence in the start of the season. Daily precipitations lower than `s_thresh_dry`
+        during `s_window_dry` days are considered a dry sequence. The precipitations must be lower than `s_thresh_dry`
+        for either every day in the sequence (`s_method_dry == "per_day"`) or for the total (`s_method_dry == "total"`).
+    s_method_dry: {"per_day", "total"}
+        Method used to define a dry sequence associated with `s_window_dry`. The threshold `s_thresh_dry` is either compared
+        to every daily precipitations (`s_method_dry == "per_day"`) or to total precipitations (`s_method_dry == "total"`)
+        in the sequence `s_window_dry` days.
+    start_date_min: DayOfYearStr
+        First day of year when season can start ("mm-dd").
+    start_date_max: DayOfYearStr
+        Last day of year when season can start ("mm-dd").
+    e_thresh_dry: str
+        Threshold defining the sequence of dry days related to `e_window_dry`.
+    e_window_dry: int
+        Number of days used to define a dry sequence in the end of the season. Daily precipitations lower than `e_thresh_dry`
+        during `e_window_dry` days are considered a dry sequence. The precipitations must be lower than `e_thresh_dry`
+        for either every day in the sequence (`e_method_dry == "per_day"`) or for the total (`e_method_dry == "total"`).
+    e_method_dry: {"per_day", "total"}
+        Method used to define a dry sequence associated with `e_window_dry`. The threshold `e_thresh_dry` is either compared
+        to every daily precipitations (`e_method_dry == "per_day"`) or to total precipitations (`e_method_dry == "total"`)
+        in the sequence `window_dry` days.
+    end_date_min: DayOfYearStr
+        First day of year when season can end ("mm-dd").
+    end_date_max: DayOfYearStr
+        Last day of year when season can end ("mm-dd").
+    freq : str
+      Resampling frequency.
+    coord : {None, "default", name of a DateTimeAccessor}
+        If `coord == "default"`, the function returns values along `dim` instead of indexes (`coord is None`).
+        If `dim` has a datetime dtype, `coord` can also be a str of the name of the
+        DateTimeAccessor object to use (ex: 'dayofyear').
+
```

Comment:
`coord`  can't be a `str | bool | None` like in `rl.season` (unless I add a new input kind.

In `rl.season`, there are 3 possibilities:
* `coord==False` or `coord is None` : output index, an integer
* `coord==True` : output the related time
* or, for example, more specific: `coord=="dayofyear"`

With a `str | bool`, I achieve the 3 cases above with:
* `coord is None` : output index, an integer
* `coord=="default"` : output the related time
* or, for example, more specific: `coord=="dayofyear"`

Is that an acceptable/clear workaround?

Reply:
I think this is fine. You agree @aulemahal ?

Reply:
Well, I agree that your annotation and "default" makes sense. However, I'm not sure if we really need to give this option to the user?
What are the use cases for something else that a `doy` ?
Disclaimer : I haven't looked at this PR more than this comment and section for now.

Reply:
Not really a big motivation, I just instinctively wanted to give as much possibility as in `rl.season`.  

I see by comparison that `frost_free_season_end` returns only a `doy`, same for `growing_season_end`, all the snow_indices, etc. So might as well follow your suggestion and make this uniform.

Reply:
Well, I don't have such strong opinions about what the indice does, but for consistency, xclim indicators always return the same units (i.e. they don't depend on parameters). So if `coord` stays in the function, the indicator should at least inject "dayofyear".

Reply:
I changed it as you suggested, the function is already complicated enough, I am gladly removing some lines.